homeassistant/components/playstation_network/coordinator.py

```diff
@@ -143,13 +144,26 @@ class PlaystationNetworkGroupsUpdateCoordinator(
 
     async def update_data(self) -> dict[str, GroupDetails]:
         """Update groups data."""
-        return await self.hass.async_add_executor_job(
-            lambda: {
-                group_info.group_id: group_info.get_group_information()
-                for group_info in self.psn.client.get_groups()
-                if not group_info.group_id.startswith("~")
-            }
-        )
+        try:
+            return await self.hass.async_add_executor_job(
+                lambda: {
+                    group_info.group_id: group_info.get_group_information()
+                    for group_info in self.psn.client.get_groups()
+                    if not group_info.group_id.startswith("~")
+                }
+            )
+        except PSNAWPForbiddenError as e:
+            error = json.loads(e.args[0])
+            _LOGGER.info(
+                "Unable to retrieve group chats for %s from PlayStation Network "
+                "due to permissions: %s. This does not affect other integration features",
+                self.config_entry.title,
+                error["error"]["message"],
+            )
+            await self.async_shutdown()
```

Comment:
The title of the PR indicates 'once during startup', but the changes are in update_data with a 3h interval? Or don't parental controls (or other privacy settings) apply when the connection is already setup? I.e. existing API connections don't apply.

I think your commit on the logger provides the user with better feedback, but if indeed it's only once than I would suggest using `warning` like you had initially but add some additional instruction or point to documentation?

`(...) This does not affect other integration features, adjust your PSN settings and reload the integration to re-enable group chats` or something to that degree?

Reply:
I interpret it as 'will run at start & update every 3h' unless you Forbidden (for whatever reason) then this coordinator is shutdown. So we assume that when we have the API connection setup it's kept up to date every 3h - as the API connection won't be blocked.

So the PR is more 'ignore parental or other privacy changes' on startup/restart. Or if PSN sometime changes that parental/privacy applies to active sessions it will just get shut.